tests/integration_tests/datasources/test_azure.py

```diff
@@ -45,3 +48,70 @@ def test_azure_eject(session_cloud: IntegrationCloud):
                 session_cloud.cloud_instance.delete_image(snapshot_id)
         else:
             _check_for_eject_errors(instance)
+
+
+def parse_resolvectl_dns(output: str) -> dict:
+    """Parses the output of 'resolvectl dns'.
+
+    >>> parse_resolvectl_dns(
+    ...    "Global:",
+    ...    "Link 2 (eth0): 168.63.129.16",
+    ...    "Link 3 (eth1): 168.63.129.16",
+    ... )
+    {'Global': '',
+     'Link 2 (eth0)': '168.63.129.16',
+     'Link 3 (eth1)': '168.63.129.16'}
+    """
+
+    parsed = dict()
+    for line in output.splitlines():
+        if line.isspace():
+            continue
+        splitted = line.split(":")
+        k = splitted.pop(0).strip()
+        v = splitted.pop(0).strip() if splitted else ""
+        parsed[k] = v
+    return parsed
+
+
+@pytest.mark.skipif(PLATFORM != "azure", reason="Test is Azure specific")
+@pytest.mark.skipif(
+    CURRENT_RELEASE < BIONIC, reason="Easier to test on Bionic+"
+)
+def test_azure_multi_nic_setup(
+    setup_image, session_cloud: IntegrationCloud
+) -> None:
+    """Integration test for https://warthogs.atlassian.net/browse/CPC-3999.
+
+    Azure should have the primary NIC only route to DNS.
+    Ensure other NICs do not have route to DNS.
+    """
+    us = datetime.datetime.now().strftime("%f")
+    rg_params = AzureParams(f"ci-test-multi-nic-setup-{us}", None)
+    nic_one = AzureCreateParams(f"ci-nic1-test-{us}", rg_params.name, None)
+    nic_two = AzureCreateParams(f"ci-nic2-test-{us}", rg_params.name, None)
+    with session_cloud.launch(
+        launch_kwargs={
+            "resource_group_params": rg_params,
+            "network_interfaces_params": [nic_one, nic_two],
+        }
+    ) as client:
+        _check_for_eject_errors(client)
+        if CURRENT_RELEASE.series == "bionic":
```

Comment:
In the interest of consistency with the `skipif` decorator above, could you replace this predicate with `CURRENT_RELEASE == BIONIC`?